### ISSUE TITLE
Current implementation of templates is designed to fail

### ISSUE THREAD

**Issue by BlaM** [nategood/httpful#175]
I like httpful, but there is one component that makes using the whole library feel itchy: The current implementation of templates is designed to fail as soon as you use httpful to communicate with more than one API: Templates are "global".

Scenario:

- Two APIs. We try to connect to both of them through httpful.
- Before calling API A we use the templating system to modify some defaults.
- We call API A - everything works as expected
- Then we call API B.
- Everything explodes because the defaults from API A don't fit.

Instead of 

    Request::ini($template);

wouldn't it be better to have something like 

    $r = Request::get($url)->using($template)->...

?

**Comment by nategood** [comment]
I agree here. The next major revision of the library will get away from much of the static global state, in particular when it comes to templates.

**Comment by apuredol** [comment]
Hi! great suggestion, any news on the implementation? any alternative solution meanwhile?

**Comment by nategood** [comment]
Yes. I agree this is a major short coming. I identified this as the primary
goal for the next major release.
